src/net/mod.rs

```diff
@@ -904,6 +910,92 @@ pub enum MemPoolSyncData {
     TxTags([u8; 32], Vec<TxTag>),
 }
 
+/// Make QualifiedContractIdentifier usable to the networking code
+#[derive(Debug, Clone, PartialEq)]
+pub struct ContractId(QualifiedContractIdentifier);
+impl ContractId {
+    pub fn new(addr: StacksAddress, name: ContractName) -> ContractId {
+        let id_addr = StandardPrincipalData(addr.version, addr.bytes.0);
+        ContractId(QualifiedContractIdentifier::new(id_addr, name))
+    }
+
+    pub fn address(&self) -> StacksAddress {
+        StacksAddress {
+            version: self.0.issuer.0,
+            bytes: Hash160(self.0.issuer.1.clone()),
+        }
+    }
+
+    pub fn name(&self) -> ContractName {
+        self.0.name.clone()
+    }
+
+    pub fn parse(txt: &str) -> Option<ContractId> {
+        QualifiedContractIdentifier::parse(txt)
+            .ok()
+            .map(|qc| ContractId(qc))
+    }
+}
```

Comment:
What we do elsewhere in the code is add a trait for extensions, which allows you to not need to wrap the object in a new struct. Extension traits are slightly more idiomatic, but probably more importantly, we should strive to only have one way of implementing extensions in our codebase, so I'd prefer if this were implemented via an extension trait.

Reply:
Fixed